src/vault/auth/kubernetes.clj

```diff
@@ -0,0 +1,69 @@
+(ns vault.auth.kubernetes
+  "The /auth/kubernetes endpoint manages Kubernetes authentication
+  functionality.
+
+  Reference: https://www.vaultproject.io/api-docs/auth/kubernetes"
+  (:require
+    [vault.client.http :as http]
+    [vault.client.proto :as proto]
+    [vault.util :as u])
+  (:import
+    vault.client.http.HTTPClient))
+
+
+(def default-mount
+  "Default mount point to use if one is not provided."
+  "kubernetes")
+
+
+(defprotocol API
+  "The Kubernetes endpoints manage Kubernetes authentication functionality."
+
+  (with-mount
+    [client mount]
+    "Return an updated client which will resolve calls against the provided
+    mount instead of the default. Passing `nil` will reset the client to the
+    default.")
+
+  (login
+    [client params]
```

Comment:
What motivated the choice of a `params` map here, rather than positional arguments as used in the `userpass` and `github` login methods? Looking at the API docs, this only takes the two arguments, both of which are required.

Reply:
Mainly the feeling that there wasn't a natural order that came to mind for the params. I'm happy to change to `[client role jwt]` though since you're right - these are the only arguments and they're both required.

Reply:
The `role` then `jwt` ordering makes sense to me, as a parallel to `user` then `password`. Or from another perspective, a path-level argument followed by a request-body argument. :+1: